### PR TITLE
Fix known hosts check by adding check for IP from ssh config

### PR DESCRIPTION
**Pull request by janvarhol**
ssk-keyscan has bug as well that was fixed in 2019, preventing it
from returning other than RC 0 so we are not relying on it anymore

### CODE REVIEW

File: lib/lib.sh [hunk 1]
```diff
@@ -2134,7 +2141,23 @@ brdexec_repair_missing_known_hosts () {
           BRDEXEX_MISSING_KNOWN_HOSTS_LINE_NUMBER="$(grep -n ${BRDEXEX_MISSING_KNOWN_HOSTS_SERVER} ~/.ssh/known_hosts | head -n 1 | awk -F ":" '{print $1}')"
 
           ### I am very proud of the following line, but don't remember why!
+	  if [ ! -f known_hosts_lock_file ]; then
```
Comment (janbaksay) [inline lib/lib.sh]: do we want to touch lockfile in current directory, that can be possibly ro? Isn't /tmp better place?

Comment (janvarhol) [reply lib/lib.sh]: https://github.com/janvarhol/broadexec/pull/33/commits/26a2e239456a9b81ab318cc933367758c7504641 better? :)

Comment (janbaksay) [reply lib/lib.sh]: much better! thanx

File: lib/lib.sh [hunk 2]
```diff
@@ -2133,8 +2140,24 @@ brdexec_repair_missing_known_hosts () {
...
+	      if [ "${BRDEXEX_KNOWN_HOSTS_LOCK_COUNTER}" -gt "${BRDEXEX_KNOWN_HOSTS_LOCK_COUNTER_MAX}" ]; then
+		log "${BRDEXEX_MISSING_KNOWN_HOSTS_SERVER} Unable to obtain lock for known hosts lockfile" 1
+		brdexec_display_output "  Warning: Unable to obtain lock when adding ${BRDEXEX_MISSING_KNOWN_HOSTS_SERVER_NAME} to the known hosts" 2
+		return 1
+	      fi
+	    done
+	    touch "/tmp/${BRDEXEC_RUNID}_known_hosts_lock_file"
+	  fi
+	  ### I am very proud of the following line, but don't remember why!
           awk -v linenumber="${BRDEXEX_MISSING_KNOWN_HOSTS_LINE_NUMBER}" -v hostname="${BRDEXEX_MISSING_KNOWN_HOSTS_SERVER_NAME}" '{if(NR==linenumber){$1=$1","hostname; print}else{print}}' ~/.ssh/known_hosts > ${BRDEXEX_MISSING_KNOWN_HOSTS_TEMP} && mv ${BRDEXEX_MISSING_KNOWN_HOSTS_TEMP} ~/.ssh/known_hosts
```
Comment (janbaksay) [inline lib/lib.sh]: maybe feature of ssh-keyscan can ease the work:

janny@janny-pc:~/work/broadexec$ ssh-keyscan -T 10 centos7
getaddrinfo centos7: Name or service not known
getaddrinfo centos7: Name or service not known
getaddrinfo centos7: Name or service not known

but this directly adds IP,hostname to all lines:
janny@janny-pc:~/work/broadexec$ ssh-keyscan -T 10 192.168.121.207,centos7
# 192.168.121.207:22 SSH-2.0-OpenSSH_7.4
192.168.121.207,centos7 ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC5hjmKD+2KWvxClvjBihO3v0IX5xHDuDp2ma1Q8098ZyuW6dkCRuunPyR4jaXau37Crz8jxY8gk/mc2o05tmDs20skfq3hQJVy36CAUVuZ5Pmxlq3Un0QoD/jkTPmB+FNfCfNxo2Tu49bOrGMqUH9ySjRxICpYLtiU+98ivKxTqsxpnE8HxV4Ole6EJeS8GPi7oIUY4f+cpmtDVYcKXfhFgnRhYkFKr0Al2v4K701S+2/EwX61FOzlOxjOH039IjRsBLa8tb3iFfnHxTOAzcMm7rpAK08IA+fTVIlsTbv2OhSemoBS5BTo3nUyXXUaRz5Y665RPX0Qt+4Mv9RKS4uz
# 192.168.121.207:22 SSH-2.0-OpenSSH_7.4
192.168.121.207,centos7 ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBGwGvg4s+SdnbURJ6KWJF9cCOhgj+vwln8/WirtCT5fwPHEW+73QT8nb4Cd44SZAmTWZTa8hrHI+rU+hevifIJk=
# 192.168.121.207:22 SSH-2.0-OpenSSH_7.4
192.168.121.207,centos7 ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIB7xHh41FbNzoF3bzx74lqL00uDHl7pLilqF9wD5Kz7A

**Pull request merged**